### PR TITLE
Test for glibc compatibility before downloading linux binary.

### PR DESCRIPTION
**Pull request by sam-mccall**
The binary releases are statically linked except for glibc, they require
version 2.18 or later. This means we're not compatible with RHEL/Centos 6/7.
Rather than offer the download and have it mysteriously fail, treat this
like no package for the platform (but with a different message).

We detect this by running ldd --version and parsing the output (as ldd is
part of glibc). If this fails or isn't glibc, we continue with the install.

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -299,31 +313,59 @@ export async function upgrade(release: Github.Release, clangdPath: string) {
...
+  const version = new semver.Range(match[1], loose);
+  console.log('glibc is', version.raw, 'min is', min.raw);
+  return rangeGreater(min, version) ? version : null;
+}
+
+// Run a system command and capture any stdout produced.
+async function run(command: string, flags: string[]): Promise<string> {
+  const child = child_process.spawn(command, flags,
+                                    {stdio: ['ignore', 'pipe', 'ignore']});
+  var output = '';
```
Comment (hokein) [inline src/index.ts:362]: nit: var -> let

File: src/index.ts [hunk 2]
```diff
@@ -299,31 +313,59 @@ export async function upgrade(release: Github.Release, clangdPath: string) {
...
+  // Require some confirmation this is [e]glibc, and a plausible
+  // version number.
+  const match = line.match(/^ldd .*glibc.* (\d+(?:\.\d+)+)[^ ]*$/i);
+  if (!match || !semver.validRange(match[1], loose)) {
+    console.error(`Can't glibc version from ldd --version output: ${line}`);
+    return null;
+  }
+  const version = new semver.Range(match[1], loose);
+  console.log('glibc is', version.raw, 'min is', min.raw);
+  return rangeGreater(min, version) ? version : null;
```
[the remark this answers is not in the document]
Comment (sam-mccall) [reply src/index.ts:355]: rangeGreater (or overlap) returns false, so we return null here (the glibc is not old, i.e. we have no "old version").
The doc comment on this function is wrong, but I think the logic is correct.

[the remark this answers is not in the document]
Comment (hokein) [reply src/index.ts:355]: oops, I see, I misunderstood this function -- I thought it returns a version if the system glibc >= 2.18.
why not making this funciton just return the current glibc version, and comparing the version in the caller side?

Maybe worth adding a testcast to test the boundary 2.18?

[the remark this answers is not in the document]
Comment (sam-mccall) [reply src/index.ts:355]: I didn't want to make the caller deal with semver, expose rangeGreater etc. This is vaguely similar to upgrade() but the signature is admittedly uglier.

I'll add the testcase indeed.
